axelor-account/src/test/java/com/axelor/apps/account/service/financialdiscount/TestInvoiceFinancialDiscountService.java

```diff
@@ -0,0 +1,264 @@
+  protected final FinancialDiscountRepository financialDiscountRepository;
+  protected final PartnerRepository partnerRepository;
+  protected final PaymentConditionRepository paymentConditionRepository;
+  protected final CompanyRepository companyRepository;
+  protected final InvoiceFinancialDiscountService invoiceFinancialDiscountService;
+  protected final InvoiceTermService invoiceTermService;
+  protected static final LoaderHelper loaderHelper = Beans.get(LoaderHelper.class);
+
+  @Inject
+  public TestInvoiceFinancialDiscountService(
+      InvoiceRepository invoiceRepository,
+      FinancialDiscountRepository financialDiscountRepository,
+      PaymentConditionRepository paymentConditionRepository,
+      CompanyRepository companyRepository) {
+    this.invoiceRepository = invoiceRepository;
+    this.financialDiscountRepository = financialDiscountRepository;
+    this.paymentConditionRepository = paymentConditionRepository;
+    this.companyRepository = companyRepository;
+
+    RequestScoper scope = ServletScopes.scopeRequest(Collections.emptyMap());
+    try (RequestScoper.CloseableScope ignored = scope.open()) {
+      this.invoiceFinancialDiscountService = Beans.get(InvoiceFinancialDiscountService.class);
+      this.partnerRepository = Beans.get(PartnerRepository.class);
+      this.invoiceTermService = Beans.get(InvoiceTermService.class);
+    }
+  }
+
+  @BeforeAll
+  static void setUp() {
+    loaderHelper.importCsv("data/account-config-input.xml");
+  }
+
+  protected Invoice prepareInvoice(Long paymentConditionId) throws AxelorException {
+    Invoice invoice = new Invoice();
+    invoice.setInvoiceDate(LocalDate.of(2024, 1, 1));
+    invoice.setDueDate(LocalDate.of(2024, 1, 15));
+    invoice.setExTaxTotal(new BigDecimal(100));
+    invoice.setInTaxTotal(new BigDecimal(100));
+    invoice.setAmountRemaining(new BigDecimal(100));
+    invoice.setOperationTypeSelect(InvoiceRepository.OPERATION_TYPE_SUPPLIER_PURCHASE);
+    invoice.setOperationSubTypeSelect(InvoiceRepository.OPERATION_SUB_TYPE_DEFAULT);
+    invoice.setPaymentCondition(
+        paymentConditionRepository
+            .all()
+            .filter("self.importId = :importId")
+            .bind("importId", paymentConditionId)
+            .fetchOne());
+    /*invoice.setCompany(companyRepository.all()
+    .filter("self.importId = :importId")
+    .bind("importId", 1L)
+    .fetchOne());*/
+    invoice.setStatusSelect(InvoiceRepository.STATUS_DRAFT);
+    invoice.setInvoiceTermList(new ArrayList<>());
+    invoice = invoiceTermService.computeInvoiceTerms(invoice);
+    return invoice;
+  }
+
+  @Test
+  void testSetFinancialDiscountInformation() throws AxelorException {
+    Invoice invoiceMonoInvoiceTerm = prepareInvoice(1L);
+    Invoice invoiceMultiInvoiceTerm = prepareInvoice(1000L);
+
+    testSetFinancialDiscountInformationOnInvoice(invoiceMonoInvoiceTerm);
+    testSetFinancialDiscountInformationOnInvoice(invoiceMultiInvoiceTerm);
+  }
```

Comment:
Please refactor your tests. Each test must have a single assert, and test directly the return of the called method of the service.